test/jest_integration/saml_multiauth.test.ts

```diff
@@ -273,7 +274,10 @@ describe('start OpenSearch Dashboards server', () => {
       until.elementsLocated(By.xpath('/html/body/div[1]/div/header/div/div[2]')),
       20000
     );
+    await driver.sleep(30000); // TODO: REMOVE THIS DEBUGGING STATEMENT
     const windowHash = await driver.getCurrentUrl();
+    const pageResource = await driver.getPageSource(); // TODO: REMOVE THIS DEBUGGING STATEMENT
+    console.log('Current page resource are: ' + pageResource);
     console.log('windowHash: ' + windowHash);
     expect(windowHash).toEqual(urlWithoutHash);
```

Comment:
I took a quick peek at the results, it looks like the tests are all getting this value on linux `http://localhost:5601/app/home#/` - This is equivalent to `http://localhost:5601/app/home`. However, if all the tests fail that seems to indicate a consistent response.  I think we should change the expected value to `http://localhost:5601/app/home#/`?

Note: This didn't fail on windows, I think this test is excluded in the GitHub Action can you confirm?

Reply:
thats the experiment I did, and I'm writing up a long comment

Reply:
https://github.com/opensearch-project/security-dashboards-plugin/pull/1641#issuecomment-1804842802

Reply:
Btw we have disabled selenium testing on windows before, thats why it only shows on linux: https://github.com/opensearch-project/security-dashboards-plugin/actions/runs/6817812839/job/18542115063?pr=1641#step:17:4

Reply:
There is logic to wait until elements are loaded or on the screen, can we use a waiter like that to verify the URL is in the expected state?  This would be an optimal solution because it would constant waits of 20 seconds, what do you think?

Reply:
Can we not just swap the expected URL to include the # then? It seems like this is the cause of the issue according to Ryan's tests

Reply:
Hi, @peternied and @scrawfor99. Thanks for the feed back. I have switched to expected url to include the `#`.  In addition to that, the solution I'm doing right now is to add the some extra locating logics on multiple elements (see here: https://github.com/opensearch-project/security-dashboards-plugin/pull/1641/files#diff-1a6a5b9310a8736f0fa34b01c4c4fed40821d39369a82973833fa9d6d21eeed9R277), so that we can make sure the loading of the page has finished. And the result is good also: https://github.com/opensearch-project/security-dashboards-plugin/actions/runs/6829392319/job/18575432125?pr=1641.

Reply:
Can the test be robust to either containing a hash or not? The hash at the end of a url does nothing unless you put an identifier after it. When an identifier is present, it scrolls the webpage down to the element with that ID in the DOM.

Reply:
Hi @cwperks, I think the main page will always contain the hash if it finishes the loading of the entire page.

Reply:
@cwperks, I think it should be this piece: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/f822702cfc4466201f6c48d805513f5d8c82d52e/test/functional/page_objects/common_page.ts#L229-L246